src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -194,7 +195,16 @@ public static LinePositionSpan RangeToLinePositionSpan(LSP.Range range)
         public static TextSpan RangeToTextSpan(LSP.Range range, SourceText text)
         {
             var linePositionSpan = RangeToLinePositionSpan(range);
-            return text.Lines.GetTextSpan(linePositionSpan);
+
+            try
+            {
+                return text.Lines.GetTextSpan(linePositionSpan);
+            }
+            // Temporary exception reporting to investigate https://github.com/dotnet/roslyn/issues/66258.
+            catch when (FatalError.ReportAndPropagate(new Exception($"Failed GetTextSpan calculation.")))
```

Comment:
```suggestion
            catch (Exception e) when (FatalError.ReportAndPropagate(e))
```

Chatting privately, @allisonchou was concerned that if we reported the exception we'd end up reporting it twice which would make finding the right dump slightly harder, but we already have code in FatalError which prevents reporting the same exception twice.